[Feature] MFDFA: allow for q=0

DominiqueMakowski opened the pull request:
https://github.com/LRydin/MFDFA/issues/33

- [x] Implement *maxFDA*

DominiqueMakowski commented:
```python
import matplotlib
import neurokit2 as nk

signal = nk.signal_simulate(duration=10, frequency=[5, 7, 10, 14], noise=0.05)
mfdfa, info = nk.fractal_mfdfa(signal, q=np.linspace(-3, 3, 7), show=True, scale = nk.expspace(50, 500, 20))

fig = matplotlib.pyplot.gcf()
fig.set_size_inches(18.5, 10.5)
```

![image](https://user-images.githubusercontent.com/8875533/180349064-ccec302a-f5af-4c35-8668-e9303fff50e3.png)


@LRydin I tried to implement it but I suspect that something is off, from the plot as the fluctuation values for when q=0 are weird

DominiqueMakowski commented on neurokit2/complexity/fractal_dfa.py:
```diff
@@ -291,6 +291,132 @@ def fractal_dfa(
+    fluctuation = np.float_power(np.mean(np.float_power(var, q_non0 / 2), axis=1), 1 / q_non0.T)
+
+    if np.sum(is0) > 0:
+        fluc0 = np.exp(np.mean(np.log(var)))
```
what I did is that I computed the values for q=0 here and then inserted them in the fluctuation list

LRydin replied:
Could you maybe give this a try (sorry for not trying myself, I have to finish something today, I was just puzzled with the disparity in the plot. I think it has to do with a `0.5` that is missing):

```python
fluc0 = np.exp(0.5*np.mean(np.log(var)))
```

DominiqueMakowski replied:
![image](https://user-images.githubusercontent.com/8875533/181870356-41b26c6c-7bb8-4895-b772-309120aa3679.png)

ah yes that looks better!

LRydin commented:
Perfect, the result is correct, you can see that the maximum of the _singularity dimension (D)_ is `1`, and that comes to be when we have `q=0`, so that's perfect :). That was easier than I expected in the end 👍.

DominiqueMakowski commented:
I added an "experimental" feature that I wanted your opinion on: 

The fluctuations often display a non-linear pattern, and the longer the scales the more the coefficient of the linear slope is biased:

```python
import matplotlib
import neurokit2 as nk

signal = nk.signal_simulate(duration=10, frequency=[5, 7, 10, 14], noise=0.05)
_, _ = nk.fractal_dfa(signal, show=True)
```

![image](https://user-images.githubusercontent.com/8875533/183017875-f8199c38-3504-45f5-aea0-2fe433280a72.png)


Intuitively the linear fit seems off.

I added the option to detect the knee of the fluctuation pattern (using the *kneedle* algorithm), and then use that knee as a max. scale before computing the slopes. This results in much closer fits:

```python
_, _ = nk.fractal_dfa(signal, maxdfa=True, show=True)
```

![image](https://user-images.githubusercontent.com/8875533/183018171-5a4f7979-b776-4558-82ea-8215a01ab44f.png)


For multifractal, I use the max. knee (of the patterns corresponding to different qs) as the limit:



![image](https://user-images.githubusercontent.com/8875533/183018376-99bd26e8-e7c6-4c82-8a8c-2fb1f4047edb.png)


![image](https://user-images.githubusercontent.com/8875533/183018431-6455d49a-0c6b-4291-b8b4-b42d98684164.png)


Does that make sense to you?

DominiqueMakowski commented:
@LRydin I'll merge this, but still do let me know if you have any thoughts on that when you have time ;) and thanks again for your help!

LRydin commented:
OMG, I never replied to this! I'm sorry :(! Yes, this makes absolute sense. That plateau that you see on your second image is fairly normal for computer-generated data. To be best of my understanding, i.e., I've never seen a paper discuss this, the plateau appears as a limit to the standard deviation of your entire process, so, putting it simply, (MF)DFA cannot detect changes in the standard deviation because the process `signal` is "bounded" between to values (quotation marks there), so it makes much more sense to extract the flucutation function at a lag smaller than the kink/keen :)

It's great! Sorry I took forever to reply @DominiqueMakowski  :|

DominiqueMakowski commented:
No worries, thanks a lot @LRydin ;)
